File: 向量检索原理/度量方法算法实现/euclidean_distance.cpp
// euclidean_distance.cpp
#include <cmath>
#include <vector>
#include <iostream>
#include <stdexcept>

double euclidean_distance(const std::vector<double>& vec1, const std::vector<double>& vec2) {
    if (vec1.size() != vec2.size()) throw std::invalid_argument("Vectors must have the same length.");
    double sum = 0.0;
    for (size_t i = 0; i < vec1.size(); ++i) {
        sum += std::pow(vec1[i] - vec2[i], 2);
    }
    return std::sqrt(sum);
}

int main() {
    std::vector<double> vec1 = {1.0, 2.0, 3.0};
    std::vector<double> vec2 = {4.0, 5.0, 6.0};
    std::cout << "Euclidean Distance: " << euclidean_distance(vec1, vec2) << std::endl;
    return 0;
}

File: 向量检索原理/度量方法算法实现/euclidean_distance.py
# euclidean_distance.py
import math

def euclidean_distance(vec1, vec2):
    if len(vec1) != len(vec2):
        raise ValueError("Vectors must have the same length.")
    return math.sqrt(sum((a - b) ** 2 for a, b in zip(vec1, vec2)))

if __name__ == "__main__":
    vec1 = [1.0, 2.0, 3.0]
    vec2 = [4.0, 5.0, 6.0]
    print("Euclidean Distance:", euclidean_distance(vec1, vec2))
